apps/condo/domains/user/schema/SignInUserService.js

```diff
@@ -0,0 +1,236 @@
+    const ip = rawIp.split(':').pop()
+    const key = `signInUser:ip:${ip}`
+    const byIpCounter = await REDIS_GUARD.incrementDayCounter(key)
+    if (byIpCounter > MAX_ANONYMOUS_REQUESTS_BY_DAY && !IP_WHITE_LIST.includes(ip)) {
+        throw new GQLError(GQL_ERRORS.SMS_FOR_IP_DAY_LIMIT_REACHED, context)
+    }
+}
+
+const SignInUserService = new GQLCustomSchema('SignInUserService', {
+    types: [
+        {
+            access: true,
+            type: 'input SignInUserUserDataInput { phone: String, name: String, email: String, password: String, meta: JSON }',
+        },
+        {
+            access: true,
+            type: 'input SignInUserInput { dv: Int!, sender: SenderFieldInput!, confirmActionToken: ID!, userType: UserTypeType!, userData: SignInUserUserDataInput }',
+        },
+        {
+            access: true,
+            type: 'type SignInUserOutput { user: User, token: ID! }',
+        },
+    ],
+
+    mutations: [
+        {
+            access: access.canSignInUser,
+            schema: 'signInUser(data: SignInUserInput!): SignInUserOutput',
+            doc: {
+                summary: 'This mutation authorizes the user by token, after confirming the phone number.\n\n' +
+                    'If the user is not registered, then he will be created with the data that is passed in the payload (user data).\n\n' +
+                    'If the existing user is missing some fields, then these fields will be taken from the payload (user data) and updated.\n\n' +
+                    'This mutation is not available for service users!',
+                errors: Object.fromEntries(Object.entries(ERRORS).filter(([, error]) => typeof error === 'object')),
+            },
+            resolver: async (parent, args, context) => {
+                const { data } = args
+                const { confirmActionToken, userType, userData, dv, sender } = data
+
+                await checkDayRequestLimitCounters(context, context.req.ip)
+
+                const dvAndSender = { dv, sender }
+                const userInfo = {
+                    ...userData,
+                    type: userType,
+                    name: String(get(userData, 'name', '')).trim() || null,
+                    email: normalizeEmail(get(userData, 'email')) || null,
+                    isEmailVerified: false,
+                    phone: normalizePhone(get(userData, 'phone')) || null,
+                    isPhoneVerified: false,
+                }
+
+                checkDvAndSender(data, ERRORS.DV_VERSION_MISMATCH, ERRORS.WRONG_SENDER_FORMAT, context)
+
+                if (!SUPPORTED_USER_TYPES.includes(userType)) throw new GQLError(ERRORS.NOT_SUPPORTED_USER_TYPE, context)
+
+                if (!confirmActionToken) throw new GQLError(ERRORS.TOKEN_NOT_FOUND, context)
+
+                const action = await ConfirmPhoneAction.getOne(context,
+                    {
+                        token: confirmActionToken,
```

Comment:
Mb we need to rename `confirmActionToken` to `confirmPhoneActionToken`.
If we will add `ConfirmEmailAction` in the future, in this mutation will be added other `confirmAction` argument

Reply:
I thought one field for the token would be enough for us. `confirmActionToken` for both `confirmPhoneActionToken` and `confirmEmailActionToken`

And in the logic of mutation we are trying to find something out of two

We have such logic in `changePasswordWithToken`